### ISSUE TITLE
cfdisk will create overlapping partitions when using suggested settings

### ISSUE THREAD

**Issue by Ryhon0** [karelzak/util-linux#1285]
If there's free space, followed by a partition, then free space again when creating a new partition with the 1st free space selected it will suggest the starting position correctly but will incorrectly suggest the ending position to be the 2nd free space end position (possibly the last free space end position if there's more free spaces)

Example:
```
| Free space | Partition 2 | Free space |
```
When you select the first free space and use the suggested settings, it will create the following:
```
|             Still Partition 1           |
| Partition 1 | Partition 2 | Partition 1 | 
```
Partition 1 is overlapping Partition 2.

The solution to this would be to use the end position of the selected free space, not the end of the last free space

**Comment by karelzak** [comment]
What partition table type o do you use? GPT or MBR? 

It would be nice to have output from `sfdisk --dump /dev/yourdevice` and maybe also copy & past from cfdisk before and after you create the partition. Thanks!

(What I need is `Start` and `End` sectors for the partitions and free space.)

**Comment by Ryhon0** [comment]
I was using MBR but I already fixed the overlapping partitions. I created a similar setup in a virtual machine, here are the outputs of `sfdisk`:
Before:
```
root@archiso ~ # sfdisk --dump /dev/vda
label: dos
label-id: 0x1f406e6d
device: /dev/vda
unit: sectors
sector-size: 512

/dev/vda2 : start=     2099200, size=    20971520, type=83
```
After:
```
root@archiso ~ # sfdisk --dump /dev/vda
label: dos
label-id: 0x1f406e6d
device: /dev/vda
unit: sectors
sector-size: 512

/dev/vda1 : start=        2048, size=    41940992, type=83
/dev/vda2 : start=     2099200, size=    20971520, type=83
```
The `vda` disk is a 20GiB VirtIO disk
Also, this is the version I was using:
```
root@archiso ~ # cfdisk -V    
cfdisk from util-linux 2.35
```

**Comment by karelzak** [comment]
Ah, v2.35 -- I'm able to reproduce it with this version, but it seems fixed since v2.36.2 (the current version is v2.37-rc1). I'm not sure which commit fixes this issue, but it's fixed :-)

Thanks, closing.
